File: mongo.rb
require 'vertx/vertx'
require 'vertx-mongo/mongo_client'
require 'json'

MONGO_DB = ENV["MONGO_DB"]
MONGO_COLLECTION = ENV["MONGO_COLLECTION"]

MONGO_CLIENT = VertxMongo::MongoClient.create_shared($vertx, {
  connection_string: "mongodb://localhost:27017",
  db_name: MONGO_DB
})

$vertx
  .create_http_server()
  .request_handler() do |req|
    MONGO_CLIENT.find_with_options(MONGO_COLLECTION, {}, { limit: 10 }) do |_,res|
      data = JSON.generate res
      req.response().put_header("content-type", "application/json").end(data)
    end
  end
  .listen(8080)

File: mongo.js
var Vertx = require("vertx-js/vertx");
var MongoClient = require("vertx-mongo-js/mongo_client");

var mongoDb = process.env.MONGO_DB;
var mongoCollection = process.env.MONGO_COLLECTION;

var mongoClient = MongoClient.createShared(vertx, {
  connection_string: "mongodb://localhost:27017",
  db_name: mongoDb,
});

vertx
  .createHttpServer()
  .requestHandler(function (req) {
    mongoClient.findWithOptions(mongoCollection, {}, { limit: 10 }, function (res, err) {
      req
        .response()
        .putHeader("content-type", "application/json")
        .end(JSON.stringify(res));
    })
  })
  .listen(8080);
